tests/tests.rs

```diff
@@ -555,6 +555,34 @@ fn test_follow() {
     );
 }
 
+// File system boundaries (--same-file-system)
+// Limited to Unix because, to the best of my knowledge, there is no easy way to test a use case
+// file systems mounted into the tree on Windows.
+// Not limiting depth causes massive delay under Darwin, see BurntSushi/ripgrep#1429
+#[test]
+#[cfg(unix)]
+fn test_file_system_boundaries() {
+    // Can't simulate file system boundaries
+    let te = TestEnv::new(&[], &[]);
+
+    // /dev/null should exist in all sane Unixes
```

Comment:
I like the way the corresponding test in the `ignore` crate is written:
https://github.com/BurntSushi/ripgrep/blob/8892bf648cfec111e6e7ddd9f30e932b0371db68/ignore/src/walk.rs#L2041-L2073

We should simply skip this test if `/dev/null` doesn't exist OR if `/dev/null` happens to be on the same file system.

Reply:
I borrowed the Unix `device_num()` function from the linked test, the test now `return`s if one of those conditions is met.

Reply:
On second thought, I should probably credit them in a comment.